### PR TITLE
SI-4176 A repeat dose of repeated parameter type sanitization.

### PR DESCRIPTION
**Pull request by retronym**
- During eta expansion, treat parameters of type A\* as Seq[A]
- Do the same for method/class parameters as referred to by an Ident.

Also fixes SI-5967, which shows up during pattern matching.

Review by @paulp or @adriaanm.

Thanks to Ryan Hendrickson for the bug report and suggested fix.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/ast/TreeGen.scala [hunk 1]
```diff
@@ -213,17 +213,14 @@ abstract class TreeGen extends reflect.internal.TreeGen with TreeDSL {
...
+  def paramToArg(vparam: Symbol): Tree =
+    paramToArg(Ident(vparam), isRepeatedParamType(vparam.tpe))
 
-  def paramToArg(vparam: ValDef) = {
-    val arg = Ident(vparam.name)
-    if (treeInfo.isRepeatedParamType(vparam.tpt)) wildcardStar(arg)
-    else arg
-  }
+  def paramToArg(vparam: ValDef): Tree =
+    paramToArg(Ident(vparam.name), treeInfo.isRepeatedParamType(vparam.tpt))
```
Comment (adriaanm) [inline src/compiler/scala/tools/nsc/ast/TreeGen.scala:220]: uber-minor nitpick: can probably drop the `treeInfo.` qualifier

**Comment by adriaanm** [comment]
```
[scalacfork] /localhome/hudson/workspace/pr-scala-testsuite-linux-opt/src/compiler/scala/tools/nsc/ast/TreeGen.scala:220: error: type mismatch;
[scalacfork]  found   : TreeGen.this.global.Tree
[scalacfork]  required: TreeGen.this.global.Type
[scalacfork]     paramToArg(Ident(vparam.name), isRepeatedParamType(vparam.tpt))
[scalacfork]                                                               ^
[scalacfork] one error found
```

**Comment by retronym** [comment]
Yeah, the optimistic push and run didn't pan out there.

**Comment by retronym** [comment]
@adriaanm I've reverted to the original.

**Comment by retronym** [comment]
PLS REBUILD ALL

File: src/compiler/scala/tools/nsc/ast/TreeGen.scala (hunk 1 above)
Comment (adriaanm) [inline src/compiler/scala/tools/nsc/ast/TreeGen.scala:220]: ah, I was probably mistaken: both treeInfo and definitions have an isRepeatedParamType??
sorry about that

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 2]
```diff
@@ -4687,7 +4687,10 @@ trait Typers extends Modes with Adaptations with Tags {
             )
             val (tree2, pre2) = makeAccessible(tree1, defSym, pre, qual)
             // assert(pre.typeArgs isEmpty) // no need to add #2416-style check here, right?
-            stabilize(tree2, pre2, mode, pt)
+            val tree3 = stabilize(tree2, pre2, mode, pt)
```
Comment (adriaanm) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala:4690]: next time you're in the 'hood, please rechristen these vals
it's like calling your kids junior 1, 2 and 3 -- who does that anymore these days

**Comment by adriaanm** [comment]
LGTM, obsessive nitpicking aside

**Pull request merged**